### ISSUE TITLE
Zoom into circular view

### ISSUE THREAD

**Issue by tiffanydai** [TeselaGen/openVectorEditor#330]
- [x] make sure selection layer in circular zoomed view looks good (update inner radius)
- [x] make sure primers are still drawing
- [x] make sure cutsites are still drawing correctly
- [x] show base pairs at a certain zoom level
- [x] make sure seq name gets rendered nicely at different zoom levels
- [x] make sure axis tick mark labels are rendered properly
- [x] zoom should focus on selection/caret when applicable
- [x] clicking and dragging in the circular view should still work!
- [x] after zooming make sure labels are shown
- [x] when making circular view bounds smaller/larger, this shouldn't affect the bp spacing 
- [x] make sure zoomed in bps are not clickable like annotations 
- [x] make sure zoom/rotate are working properly with the full screen view!
- [x] viewing circular view in fullscreen and then navigating out should still focus the circular view
- [x] after resizing/rotating and then switching away from the circular view, upon return the labels shouldn't have a messed up rotation or disappear
- [x] row view axis ticks should still be rendering correctly
- [x] circular view zoomed in axis on origin should not include 11,21,31 (should be 10,20,30)
- [x] if a part of the selection/caret is visible, then don't force a rotation when zooming
- [x] if a large sequence is zoomed in and/or rotated and then shortened to be <50 bps, the sequence should be fully zoomed out and rotated back to 0 (since we do not allow zoom on <50 bps)
- [x] make sure external label groups are getting updated correctly on hover
- [ ] add all sorts of tests!

**Stretch goals:**
- [ ] show more detailed cutsites at a certain zoom level
- [ ] add option to force labels to be rendered externally
- [ ] internal labels should jump to be in view

Feedback from LanzaTech on OVE: Allow for zoom into plasmid map, to the extent of viewing the sequence. Will obviously look identical to the linear view, but at times the user can't get their bearings while clicking around the plasmid map and then looking over at the linear view. 

Geneious offers this feature: 
<img width="890" alt="geneious_plasmidmap" src="https://user-images.githubusercontent.com/35508772/40555181-1b96ecd8-5ffd-11e8-9e81-f48ab2978765.png">
(zooming in a little more on this view will get the nucleotides to show up)

**Comment by ericsturman** [comment]
@tnrich this is also a feature we would really like to see.  Is there an estimate for when a basic zooming feature might be implemented?

**Comment by tnrich** [comment]
@ericsturman no timeline as of yet. I think before we can give a timeline some serious thought needs to be put into how the UX would work for this feature. I'm not sure what the best way to do this is. 

Maybe you could help by giving a breakdown of how you'd want this to work as well as how other tools handle zooming in the circular view? Once we have the UX decided on I could figure out what changes we'd need to make and give you a timeline.

**Comment by tnrich** [comment]
Hey @ericsturman and @tiffanydai  I just wanted to give you a small progress report as I've been working on and off on a first pass solution for this for a little while now. Here's a video showing what I've got so far: 

![zoomdemo](https://user-images.githubusercontent.com/2730609/68057443-2022ae00-fcb3-11e9-9ac5-24b9ce4c10ce.gif)


 You can check it for yourself out on the `tnrCircularZoom` branch by running `yarn; yarn start` 

There is still a ways to go until this zoom solution is ready to be published but I thought I'd get some early feedback from you both and see what you think. All feedback is welcome :)

**Comment by tnrich** [comment]
@ericsturman I've implemented a linear view zoom feature that has now been released. You can see that on the demo page. 

I'm picking this circular view issue back up after a few years hiatus.
